src/test_graph.py

```diff
@@ -0,0 +1,331 @@
+# _*_ encoding: utf-8 _*_
+"""Test graph.py."""
+from graph  import Graph
+import pytest
+
+
+@pytest.fixture()
+def native_graph():
+    """Fixture for graph."""
+    from graph import Graph
+    test_graph = Graph()
+    test_graph.add_node("codefellows")
+    test_graph.add_node("python")
+    return test_graph
+
+
+@pytest.fixture()
+def ample_graph():
+    """Fixture that makes a full graph."""
+    from graph import Graph
+    ample_graph = Graph()
+    ample_graph.add_edge("a", "b")
+    ample_graph.add_edge("c", "d")
+    ample_graph.add_edge("c", "b")
+    ample_graph.add_edge("a", "d")
+    ample_graph.add_edge("b", "d")
+    ample_graph.add_edge("d", "a")
+    return ample_graph
+
+
+@pytest.fixture()
+def deep_cyclic_graph():
+    """Fixture that makes a deeper graph."""
+    from graph import Graph
+    ample_graph = Graph()
+    ample_graph.graph = {'A': {'B': 0, 'C': 0}, 'B': {'D': 0, 'E': 0},
+                        'C': {'D': 0, 'E': 0}, 'D': {'E': 0}, 'E': {'A': 0}}
+    return ample_graph
+
+
+@pytest.fixture()
+def deep_cyclic_graph():
+    """Fixture that makes a deeper graph."""
+    from graph import Graph
+    ample_graph = Graph()
+    ample_graph.graph = {'A': {'B': 0, 'C': 0}, 'B': {'D': 0, 'E': 0},
+                        'C': {'D': 0, 'E': 0}, 'D': {'E': 0}, 'E': {'A': 0}}
+    return ample_graph
```

Comment:
This is gonna just keep bugging me. Why is this called `ample_graph`? Did you mean `sample_graph`?